packages/headless/src/controllers/result-list/headless-result-list.ts

```diff
@@ -51,6 +51,37 @@ export function buildResultList(
     dispatch(registerFieldsToInclude(options.fieldsToInclude));
   }
 
+  let lastFetchCompleted = 0;
+  let consecutiveFetches = 0;
+  const minConsecutiveFetches = 5;
+  const maxDelayBetweenFetches = 200;
+  let errorLogged = false;
+
+  const triggerFetchMoreResult = () => {
+    if (engine.state.search.isLoading) {
+      return;
+    }
+
+    const delayBetweenFetches = Date.now() - lastFetchCompleted;
+    if (delayBetweenFetches < maxDelayBetweenFetches) {
+      consecutiveFetches++;
+      if (consecutiveFetches >= minConsecutiveFetches) {
+        lastFetchCompleted = Date.now();
+        !errorLogged &&
+          engine.logger.error(
+            `The result list method "fetchMoreResults" execution prevented because it has been triggered consecutively ${minConsecutiveFetches} times, with little delay. Please verify the conditions under which the function is called.`
+          );
+        errorLogged = true;
```

Comment:
Another option could be to update the `lastFetchCompleted` variable with `Date.now()` to avoid introducing another flag.

Reply:
We do on line 69, this is to prevent a bunch of repetitive error logs, better dev experience